ddclient.in

```diff
@@ -2276,44 +2276,37 @@ sub extract_ipv4 {
     return $ip;
 }
 
+######################################################################
+## Regex to find IPv6 in standard or compressed format. Mixed IPv6/IPv4
+## not supported. Accepts leading zeros in the address. Adapted from
+## regex in Regular Expressions Cookbook by J.Goyvaerets and S.Levithan
+######################################################################
+my $regex_ipv6 = "(?<![:.0-9A-Z])              ## Negative lookbehind for word boundry
+    ( (?:[0-9A-F]{1,4}:){7}[0-9A-F]{1,4}       ## Is in standard format
+     |(?=(?:[0-9A-F]{0,4}:){0,7}[0-9A-F]{0,4}  ## OR compressed with at most 7 colons
+      (?![:.0-9A-Z]))                          ##   and negative lookahead for boundry
+      (?:(?:[0-9A-F]{1,4}:){1,7}|:)(?:(?::[0-9A-F]{1,4}){1,7}|:)  ## and only 1 double-colon
+     |(?:[0-9A-F]{1,4}:){7}:                   ## OR compressed with 8 colons (double at end)
+     |:(?::[0-9A-F]{1,4}){7}                   ## OR compressed with 8 colons (double at front)
+    )                                          ## End of first capture group
+    (?![:.0-9A-Z])";                           ## Negative lookahead for boundry
```

Comment:
I was having problems with that during unit testing.  But looks like it might have been something else going on as when I test it now it is working.  Thanks for prompting me to look at it again.  I will update the code.